python/ml4ir/applications/ranking/features/feature_fns/rank_transform.py

```diff
@@ -38,13 +39,17 @@ def __init__(self, feature_info: dict, file_io: FileIO, **kwargs):
                 If k should be a learnable variable; will be initialized with value of k
             ignore_zero_score: bool
                 Use zero reciprocal rank for score value of 0.0
+            scale_to_one: bool
```

Comment:
Can you add why this is a needed feature.
Also, if scale_to_one is False, isn't the RR are already (0,1] ? why multiplying by k+1 would make it (0,1]?
@lastmansleeping

Reply:
It makes the linear combination weights more comparable with other [0, 1] features. Otherwise, reciprocal rank with k=60, would result in very LARGE weights due to small feature values like 1 / (60 + 1) = 0.01..

Multiplying by (k + 1) sets the max possible value to 1 as max possible reciprocal rank is 1 / (k + rank=1)

Reply:
Ok, so it's mostly re-scaling the RR spectrum, right?

Reply:
Yep.

Reply:
@lastmansleeping 
In that case, can you change the name SCALE_TO_ONE to something else as SMOOTH_RR_RANGE and perhaps adding a comment to convey that.

Reply:
But we are not smoothening the curve. We are simply "scaling to one". This is more descriptive, no?

Reply:
FWIW, this is what it is called serving side too.

Reply:
I can maybe call it `scale_range_to_one`. Does that work?

Reply:
I went with `scale_range_to_one`. Let me know if that works.